### PR TITLE
Added constraint for username password & email for signup

### PR DESCRIPTION
**Pull request by abhinav5481**
## What is the change?
Added constraints in email, password & username for signup in server-side code.

## Related issue?
issue: #163 

## How was it tested?
I have tested by entering the invalid username password & email on my local machine.

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

- [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [x] Have you made corresponding changes to the documentation?
- [x] Your submission doesn't break any existing feature.
- [x] Have you tested the code before submission?

#### Username Check: `username should be between 6 to 12 characters`
Upon entering the invalid username  an error would be returned from the server-side which would appear as
image

#### Email Check:
Similar to username, email would be checked against the email regex and the error message would be shown on the screen

#### Password Check:  ` password must contain a minimum of 8 letters, with at least a symbol, upper and lower case letters and a number`
image

Please review my code & let me know if any further changes are required.
Add a screenshot or demo video if appropriate.

### CODE REVIEW

**Comment by shudhanshu03** [comment]
Good Work @abhinav5481 . 
/reviewed

**Review by shudhanshu03 — COMMENTED** [review]
Good Work.

**Comment by shudhanshu03** [comment]
/gssoc/ 
/level2/

**Comment by shudhanshu03** [comment]
/level2/

**Comment by Pranav016** [comment]
> /merge/

not ready to merge, base branch is master, should be `develop` for any kind of code changes

**Comment by shudhanshu03** [comment]
> > /merge/
> 
> not ready to merge, base branch is master, should be `develop` for any kind of code changes

@abhinav5481 Base branch should be develop. I am adding a image hope that might help you to change .
Gs_Develop_Branch

**Comment by Pranav016** [comment]
> > > /merge/
> > 
> > 
> > not ready to merge, base branch is master, should be `develop` for any kind of code changes
> 
> @abhinav5481 Base branch should be develop. I am adding a image hope that might help you to change .
> Gs_Develop_Branch

good work

File: app.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ if(process.env.NODE_ENV !== 'production'){
 }
 
 //Connecting to Mongo Database using ODM Mongoose-
-const URL = process.env.URL;
+const URL = 'mongodb://localhost:27017/blogDB'
+// process.env.URL;
```
Comment (Pranav016) [inline app.js:33]: Revert this back, localhost is only to run it on local machine

**Review by Pranav016 — CHANGES_REQUESTED** [review]

**Comment by Pranav016** [comment]
Closing for now, reopen if required.

File: app.js (hunk 1 above)
Comment (abhinav5481) [reply app.js:33]: Okay.
I will do it and commit it back

**Pull request closed without merging**